extension-style/src/main/java/com/mapbox/maps/extension/style/StyleExtensionImpl.kt

```diff
@@ -11,7 +11,10 @@ import com.mapbox.maps.extension.style.terrain.generated.Terrain
 /**
  * The concrete implementation of style extension.
  */
-class StyleExtensionImpl private constructor(builder: Builder) : StyleContract.StyleExtension {
+class StyleExtensionImpl private constructor(
+  builder: Builder,
+  override var resourceCount: Int
```

Comment:
nit, can be val

Reply:
it could also just be a method to just count the sources/layers/light/images etc. once constructed.

Reply:
we decrease and increase it directly from `MapboxMap`, so no 😄

Reply:
why do we need it that way? I think the mapboxmap can use a var locally.

Reply:
consider that the style extension can be applied to again when the style is changed.

Reply:
current approach will handle it ok, but I like your more, will do it your way, I guess, thanks!

Reply:
Applied.